File: youqugou-interface/src/main/java/cn/itcast/core/service/BrandService.java
package cn.itcast.core.service;

import cn.itcast.core.entity.PageResult;
import cn.itcast.core.pojo.good.Brand;

import java.util.List;
import java.util.Map;

public interface BrandService {
    List<Map> selectAllBrandMap();

    PageResult selectPageBrand(Integer pageNum, Integer pageSize);

    int insertBrand(Brand brand);

    Brand selectBrandById(Long id);

    int updateBrand(Brand brand);

    int deleteBrandByIds(Long[] ids);

    PageResult searchBrand(Integer pageNum, Integer pageSize, Brand brand);

    /*贾运通*/
    void updateStatus(Long[] ids, String status);

}

File: youqugou-web-shop/src/main/webapp/js/service/brandService.js
app.service("brandService", function ($http) {
    this.searchBrand = function (currentPage, itemsPerPage,searchEntity) {
        return $http.post("../brand/searchBrand.do?pageNum=" + currentPage
            + "&pageSize=" + itemsPerPage, searchEntity);
    }

    this.selectPageBrand = function (currentPage, itemsPerPage) {
        return $http.get("../brand/selectPageBrand.do?pageNum=" + currentPage
            + "&pageSize=" + itemsPerPage)
    }

    this.insertBrand = function (entity) {
        return $http.post("../brand/insertBrand.do", entity)
    }

    this.updateBrand = function (entity) {
        return $http.post("../brand/updateBrand.do", entity)
    }

    this.selectBrandById = function (id) {
        return $http.get("../brand/selectBrandById.do?id=" + id)
    }

    this.deleteBrandByIds = function (ids) {
        return $http.post("../brand/deleteBrandByIds.do", ids)
    }

    this.selectAllBrandMap = function () {
        return $http.get("../brand/selectAllBrandMap.do");
    }
});